pkg/operation/botanist/health_check.go

```diff
@@ -187,9 +182,22 @@ func (b *HealthChecker) checkRequiredDaemonSets(condition gardencorev1beta1.Cond
 		actualNames.Insert(object.Name)
 	}
 
-	if missingNames := requiredNames.Difference(actualNames); missingNames.Len() != 0 {
-		c := b.FailedCondition(condition, "DaemonSetMissing", fmt.Sprintf("Missing required daemon sets: %v", missingNames.List()))
-		return &c
+	return b.checkRequiredResourceNames(condition, requiredNames, actualNames, "DaemonSetMissing", "Missing required daemon sets")
+}
+
+var relevantConditions = map[resourcesv1alpha1.ConditionType]struct{}{
+	resourcesv1alpha1.ResourcesApplied: struct{}{},
+	resourcesv1alpha1.ResourcesHealthy: struct{}{},
+}
+
+// CheckManagedResource checks the conditions of the given managed resource and reflects the state in the returned condition.
+func (b *HealthChecker) CheckManagedResource(condition gardencorev1beta1.Condition, mr *resourcesv1alpha1.ManagedResource) *gardencorev1beta1.Condition {
+	for _, cond := range mr.Status.Conditions {
```

Comment:
Should we also check, if `observedGeneration` of the mr is up-to-date (like [here](https://github.com/gardener/gardener-resource-manager/blob/master/pkg/health/health.go#L46))?

Reply:
Yes, actually a good idea. The question is though what we should do in such a case? Is it helpful to report an "outdated" message instead of the actual condition?

Reply:
Yes, I would probably go for an "outdated" message as this has (probably) no correlation to a single condition, right?

Reply:
Sure, let's proceed like that 👍